nextest-runner/src/cargo_config/target_triple.rs

```diff
@@ -273,6 +298,62 @@ impl TargetTriple {
     }
 }
 
+/// Cargo argument for downstream commands.
+///
+/// If it is necessary to run a Cargo command with a target triple, this enum provides the right
+/// invocation. Create it with [`TargetTriple::to_cargo_target_arg`].
+#[derive(Debug)]
+pub enum CargoTargetArg {
+    /// The target triple is a builtin.
+    Builtin(String),
+
+    /// The target triple is a JSON file at this path.
+    Path(Utf8PathBuf),
+
+    /// The target triple is a temporary JSON file.
+    TempFile(NamedUtf8TempFile),
+}
+
+impl CargoTargetArg {
+    fn from_custom_json(json: &str, source: TargetTripleSource) -> Result<Self, TargetTripleError> {
+        // Extract the JSON to a temporary file. Cargo requires that the file name end with `.json`.
+        let f = camino_tempfile::Builder::new()
+            .prefix("nextest-custom-target-")
+            .suffix(".json")
+            .rand_bytes(5)
+            .tempfile()
```

Comment:
This isn't correct -- we need to make a temporary *directory*, and then a file with the right name within it.

Reply:
Done.